### ISSUE TITLE
Feature: add Vuetify v2 support

### ISSUE THREAD

**Issue by MichaelHipp** [wailsapp/wails#226]
**Description**

Currently Wails supports Vue v2 with Vuetify v1.5. Vuetify v2.0 is the latest release with many improvements and compelling features. This issue is to propose a change to the Vuetify templates to support Vuetify v2.0.

_Vuetify v2 and Wails are important to my current projects so I am volunteering to work on this unless someone else has first dibs._ 

If this is acceptable I'd like to request a decision on one of two alternatives:

**Alternative #1 :**
Modify the existing "Vuetify Basic" template to use v2 instead of v1.5. Rename as follows:

name: "Vuetify Basic"  ==> "Vuetify2/Webpack Basic"
shortdescrption: "Vuetify + Webpack" ==> "A basic Vuetify2/WebPack4 template"

There would no longer be support in Wails for Vuetify v1.5.

**Alternative #2 :**
Create a new template (based on the existing v1.5 template) and modify it for v2 support. Leave the existing v1.5 template in place with somewhat more descrption:

v1.5:
name: "Vuetify Basic"  ==> "Vuetify1.5/Webpack Basic"
shortdescrption: "Vuetify + Webpack" ==> "A basic Vuetify1.5/WebPack4 template"

v2.0:
name: "Vuetify Basic"  ==> "Vuetify2/Webpack Basic"
shortdescrption: "Vuetify + Webpack" ==> "A basic Vuetify2/WebPack4 template"

tl'dr: there would be two (2) Vuetify templates for versions 1.5 & 2.0, respectively.

This might be appropriate as Vuetify v1.5 is still a supported version receiving regular updates and is widely used. But the obvious downside is that it adds "clutter" to the list of available templates.

In both alternatives I'm proposing to make the name & shortdescrption somewhat more informative in the model of the Vue2 template, which reads as follows:

name: "Vue2/Webpack Basic",
shortdescription: "A basic Vue2/WebPack4 template",

**System Details**

| Name   | Value |
| ----- | ----- |
| Wails Version | v0.17.14-pre |
| Go Version    | go1.13 |
| Platform      | linux |
| Arch          | amd64 |
| GO111MODULE   | on |
| GCC           | 8.3.0 |
| Npm           | v12.9.1 |
| Node          | 6.10.2 |

**Comment by bh90210** [comment]
my vote is for alternative 1 :+1: 
from our perspective alternative 2 adds future maintaining weight without much use for the user, also since #209 is going to get implemented, eventually if needed special builds with specific versions etc can be implemented and used by anyone for anyone so _our_ strategy maybe should focus on just the latest - which is hard already?

My suggestion is always to provide good enough documentation so users can perform such changes (for example downgrade to v1.5) easier/faster.

**Comment by leaanthony** [comment]
Thanks for opening this Michael. On one hand I agree with Byron that we should simply update the V1 template, but I also want to make sure it's not our most popular template ever and that everyone uses it. As we currently have no telemetry and don't track metrics like that, we have to assume some people have a need for it. So my proposal is this: let's make it a fresh template, but we sunset v1. That'll give people the chance to upgrade or migrate. 
Remember, it only affects project generation. The interface with Wails will be via the `@wailsapp/runtime` module so even existing v1 projects will be able to get those updates.

TL;DR
* Create a new template as you propose (I'm happy with the wording btw)
* We'll issue a warning for the v1 template for v0.18.0 and deprecate it in v1.0.0.

**Comment by MichaelHipp** [comment]
FYI: You can track progress here: https://github.com/MichaelHipp/wails/tree/vuetify2-support

**Comment by leaanthony** [comment]
Hi @MichaelHipp. We're about to feature freeze for v1.0.0 so I'm going to close this issue for now but have listed it on the https://github.com/wailsapp/wails/wiki/Post-V1-Enhancements page. This will be one of the first things we tackle post v1
